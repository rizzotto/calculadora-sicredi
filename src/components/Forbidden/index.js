import React from "react";

import { makeStyles, useMediaQuery } from "@material-ui/core";
import PropTypes from "prop-types";

import Logo from "../../assets/LogoSicrediLogin.svg";
import LogoMobile from "../../assets/Sicredi.svg";
import theme from "../../styles/customMuiTheme";
import Link from "../Link";

const useStyles = makeStyles({
    container: {
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        height: "100vh",
        padding: 30,
    },
    content: {
        display: "flex",
        alignItems: "center",
        justifyContent: "space-evenly",
        height: "100vh",
        flexDirection: "column",
        width: "100%",
        maxWidth: 600,
    },
    text: {
        fontSize: 36,
        color: theme.palette.primary.main,
    },

    textMobile: {
        fontSize: 25,
        color: theme.palette.primary.main,
    },
});

function Forbidden({ role }) {
    const classes = useStyles();
    const isMobile = useMediaQuery(theme.breakpoints.down("sm"));

    function handleOnClick() {
        if (role === "associate") {
            return "/";
        } else if (role === "coop") {
            return "/coop";
        } else {
            return "/admin";
        }
    }

    return (
        <div className={classes.container}>
            <div className={classes.content}>
                <img
                    alt="Logo"
                    src={isMobile ? LogoMobile : Logo}
                    className={classes.image}
                />
                <h2 className={isMobile ? classes.textMobile : classes.text}>
                    Você não tem acesso a essa página
                </h2>
                <Link color="white" to={handleOnClick}>
                    HOME
                </Link>
            </div>
        </div>
    );
}

Forbidden.propTypes = {
    role: PropTypes.string.isRequired,
};

export default Forbidden;
