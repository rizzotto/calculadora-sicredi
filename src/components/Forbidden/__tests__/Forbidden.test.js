import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import Forbidden from "../index.js";

it("renders Forbidden correctly", () => {
    const tree = renderer
        .create(
            <Router>
                <Forbidden />
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
