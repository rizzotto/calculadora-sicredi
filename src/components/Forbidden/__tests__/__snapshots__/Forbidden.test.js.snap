// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Forbidden correctly 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    className="makeStyles-content-2"
  >
    <img
      alt="Logo"
      src="LogoSicrediLogin.svg"
    />
    <h2
      className="makeStyles-text-3"
    >
      Você não tem acesso a essa página
    </h2>
    <a
      className="makeStyles-white-6"
      href="/admin"
      onClick={[Function]}
    >
      HOME
    </a>
  </div>
</div>
`;
